python_modules/dagster/dagster/components/resolved/model.py

```diff
@@ -83,6 +83,24 @@ def from_model(fn: Callable[["ResolutionContext", Any], Any], **kwargs):
         """Resolve this field by invoking the function which will receive the entire parent model."""
         return Resolver(ParentFn(fn), **kwargs)
 
+    @staticmethod
+    def union(*resolvers: "Resolver"):
+        def _resolve_fn(context: "ResolutionContext", field_value: Any):
+            for r in resolvers:
+                if not r.model_field_type or isinstance(field_value, r.model_field_type):
```

Comment:
how do we handle the instance check here if `model_field_type` isn't available (e.g. on a `_is_implicitly_resolved_type` type?)

I wonder if we'll want to build up an equivalent to the `_wrap`ped `model_field_type` even in the implicit case as part of `_dig_For_resolver` so that we can pass the type to instance-check against in this fn

Reply:
can we build up a mapping at [1] of `arg`  -> `_dig_for_resolver` result and use that somehow? I wonder if we need a richer "path" structure than the current list of enum entries